src/System.Management.Automation/FormatAndOutput/common/Utilities/MshObjectUtil.cs

```diff
@@ -290,17 +291,20 @@ internal static string SmartToString(PSObject so, PSPropertyExpressionFactory ex
                 // take care of the case there is no base object
                 return so.ToString();
             }
-            catch (ExtendedTypeSystemException e)
+            catch (Exception e)
```

Comment:
Consider:

```csharp
catch (Exception e) when (e is ExtendedTypeSystemException || e is InvalidOperationException)
{
    if (formatErrorObject != null)
    {
        formatErrorObject.sourceObject = so;
        formatErrorObject.exception = e;
    }

    return string.Empty;
}
```

Reply:
Alternatively, since the two errors don't really have a common provenance, it might make sense for them to have different code paths:

```csharp
catch (ExtendedTypeSystemException etsEx)
{
    if (formatErrorObject != null)
    {
        formatErrorObject.sourceObject = so;
        formatErrorObject.exception = e;
    }

    return string.Empty;
}
catch (InvalidOperationException invalidOpEx)
{
    // Handling appropriate for the InvalidOperationException
}
```

An `InvalidOperationException` is a pretty generic and high-level exception though and one that's seldom caught, so I wonder about catching them in general.

I would be a lot more comfortable knowing more about when and why this exception is thrown to understand what handling should be done other than just sitting on it.

Reply:
There isn't different handling for the two exception types, so will go with first proposal